worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala

```diff
@@ -562,38 +591,69 @@ class PushDataHandler(val workerSource: WorkerSource) extends BaseMessageHandler
             workerSource.incCounter(WorkerSource.REPLICATE_DATA_CREATE_CONNECTION_FAIL_COUNT)
             logError(
               s"PushMergedData replication failed caused by unavailable peer for partitionLocation: $location")
-            callbackWithTimer.onFailure(
+            pushMergedDataCallback.onFailure(
               new CelebornIOException(StatusCode.PUSH_DATA_CREATE_CONNECTION_FAIL_REPLICA))
             return
           }
 
           // Handle the response from replica
           val wrappedCallback = new RpcResponseCallback() {
             override def onSuccess(response: ByteBuffer): Unit = {
+              val replicaReason = response.get()
+              try {
+                val pushMergedDataResponse: PbPushMergedDataUnsuccessfulPartitionInfo =
+                  TransportMessage.fromByteBuffer(
+                    response).getParsedPayload[PbPushMergedDataUnsuccessfulPartitionInfo]()
+                pushMergedDataCallback.unionSplitOrFailedPartitions(
+                  pushMergedDataResponse.getSplitOrFailedPartitionIndexesList,
+                  pushMergedDataResponse.getStatusCodesList)
+              } catch {
+                case e: CelebornIOException =>
+                  pushMergedDataCallback.onFailure(e)
+                  return
+                case e: IllegalArgumentException =>
+                  pushMergedDataCallback.onFailure(new CelebornIOException(e))
+                  return
+              }
               Try(Await.result(writePromise.future, Duration.Inf)) match {
-                case Success(_) =>
+                case Success(result) =>
+                  var index = 0
+                  var indexOfUnWritableIndexes = 0
+                  while (index < result.length) {
+                    if (indexOfUnWritableIndexes < unWritableIndexes.length && index == unWritableIndexes(
+                        indexOfUnWritableIndexes)) {
+                      indexOfUnWritableIndexes += 1
+                    } else {
+                      if (result(index) != StatusCode.SUCCESS) {
+                        pushMergedDataCallback.addSplitOrFailedPartition(index, result(index))
+                      }
+                    }
+                    index += 1
+                  }
```

Comment:
Here, I think we should to check if any partition need Split, then we need return  HARD_SPLIT status code

Reply:
If any partition need Split, we will invoke pushMergedDataCallback.addSplitOrFailedPartition to collect it. When invoke pushMergedDataCallback.onSuccess and the splitOrFailedPartitionStatuses is not empty, the final status will be HARD_SPLIT